content/blog/10-pearls-with-azure-functions-in-pulumi/index.md

```diff
@@ -0,0 +1,361 @@
+---
+title: "10 Pearls With Azure Functions in Pulumi"
+date: 2019-08-15
+draft: true
+meta_image: meta.png
+authors: ["mikhail-shilkov"]
+tags: ["Azure", "Serverless"]
+meta_desc: "In this post, we'll take a look at 10 pearls&mdash;bite-sized code snippets&mdash;that demonstrate using Pulumi to build serverless applications with Azure Functions and infrastructure as code."
+---
+
+In this post, we'll take a look at 10 "pearls"&mdash;bite-sized code snippets&mdash;that demonstrate using Pulumi to build serverless applications with Azure Functions and infrastructure as code. These pearls are organized into four categories, each demonstrating a unique scenario:
+
+- **Function App Deployment**: Take an existing Azure Functions application in your language of choice and deploy it with infrastructure as code.
+- **HTTP Functions as Callbacks**: Mix TypeScript or JavaScript functions right into the infrastructure definition to produce concise, strongly-typed, self-contained, serverless HTTP endpoints.
+- **Cloud Event Handling**: Leverage the whole variety of event sources available to Azure Functions with lightweight and intuitive event handlers.
+- **Data Flows with Function Bindings**: Pull extra data into the functions and push results out with function bindings&mdash;declarative connectors to Azure services.
+<!--more-->
+
+Here is a complete index of the pearls including associated steps, in case you want to jump to a specific one:
+
+[**Function App Deployment**](#function-app-deployment)
+
+- [Deploy a Function App written in .NET or any other supported runtime](#1-deploy-a-net-function-app)
+- [Configure Functions to run on an Elastic Premium Plan](#2-run-functions-using-an-elastic-premium-plan)
+
+[**HTTP Functions as Callbacks**](#http-functions-as-callbacks)
+
+- [Define Node.js Functions as inline callbacks](#3-define-node-js-functions-as-inline-callbacks)
+- [Implement REST APIs as multiple Functions](#4-rest-apis-as-multiple-functions)
+- ["Warm" the Functions to avoid Cold Starts](#5-function-warming-with-a-timer-function)
+
+[**Cloud Event Handling**](#cloud-event-handling)
+
+- [Process events from Azure Event Hub](#6-process-events-from-azure-event-hubs)
+- [Subscribe to Blob creation with Azure Event Grid](#7-subscribe-to-azure-event-grid)
+- [Run a Function every time an Azure resource is modified](#8-respond-to-resource-level-events)
+
+[**Data Flows with Function Bindings**](#data-flows-with-function-bindings)
+
+- [Push a message to a Storage Queue with an output binding](#9-output-bindings)
+- [Pull a Storage Table row with an input binding](#10-input-bindings)
+
+## Function App Deployment
+
+Azure Functions support many languages, and Pulumi doesn't limit your choice. You are free to take any existing serverless application and deploy it in an infrastructure-as-code manner.
```

Comment:
```suggestion
Azure Functions support many languages, and Pulumi supports whatever choice you make. You can take any existing serverless application and deploy it using infrastructure-as-code.
```

Reply:
Two "support" in one sentence. "Pulumi respects"? "plays along with"?